packages/proxy/lib/http/response-middleware.ts

```diff
@@ -656,6 +656,10 @@ const MaybeSendRedirectToClient: ResponseMiddleware = function () {
     return this.next()
   }
 
+  if (this.req.noPreRequestExpected) {
```

Comment:
If we're redirecting from a request that doesn't expect to have a preRequest (e.g. download links), we need to treat the redirected url as such as well.